doc/man3/SSL_write.pod

```diff
@@ -34,10 +34,7 @@ L<SSL_set_connect_state(3)> or SSL_set_accept_state()
 before the first call to a write function.
 
 If the underlying BIO is B<blocking>, the write functions will only return, once
-the write operation has been finished or an error occurred, except when a
-renegotiation take place, in which case a SSL_ERROR_WANT_READ may occur.
-This behaviour can be controlled with the SSL_MODE_AUTO_RETRY flag of the
-L<SSL_CTX_set_mode(3)> call.
+the write operation has been finished or an error occurred.
```

Comment:
I'm still thinking it is be possible to get `SSL_ERROR_WANT_READ` from `SSL_write()`. Consider the scenario where a server application call SSL_read() and receives a reneg ClientHello from the client. It goes on to write out its ServerHello etc and eventually returns with SSL_ERROR_WANT_READ (because it is now waiting to receive the client Finished). If the server application then immediately calls SSL_write(), I think it will still return SSL_ERROR_WANT_READ.

Reply:
I can not reproduce this using blocking I/O. I also really dislike that I can't figure any of this out from reading the code and that I need to test what the actual behaviour is.

Reply:
The code isn't really clear to me yet. At least when ssl3_read_bytes() is called from SSL_write(), I think we're in handshake state. To get to the part of the SSL_MODE_AUTO_RETRY we need type != rr->type and not be in handshake mode. But I might be missing various subtle things.